### PR TITLE
Feat/tracks and lessons with dynamic segnments

### PR DESCRIPTION
**Pull request by Markkos89**
## Changes

- dynamic segnments fundamentals and tracks pages
before:
image

now:
image

- using `generateStaticParams()` Next.js method to generate statically at build time lessons and tracks pages. i.e.:
image

- Left some `TODO`'s and  `DEV_NOTES` around:
image

**IMPORTANT:** _We need to populate the database with authors information and authors relationship with tracks and lessons so we can populate dynamically the author information in the `TrackLayout` component_ cc: @kempsterrrr 

related to #15 

### New Features *(required)*

Fixes #
Connects #

### CODE REVIEW

**Comment by kempsterrrr** [comment]
Can you ELI5 what is happeing in this PR please @Markkos89 ? I'm not sure what we're trying to achieve :)

**Comment by Markkos89** [comment]
> Can you ELI5 what is happeing in this PR please @Markkos89 ? I'm not sure what we're trying to achieve :)

- we are building more pages statically now.
- we are reusing Layout.tsx and Page.tsx files instead of creating one per lesson & per track.
- the markdown content is moved to a dedicated folder

This enhancements should make the app more SEO friendly 

**IMPORTANT:** Now that we are reusing the layouts we have to populate the database with authors information and create the relationship of each author with the proper track and lesson. @kempsterrrr :pray:

**Comment by Markkos89** [comment]
@kempsterrrr going to merge this

**Comment by kempsterrrr** [comment]
S H I P IT
